### PR TITLE
Example of building a container without sudo using Uproot

### LINKED ISSUE #32: Example to build a container that does not require superuser permissions
In chapter 4, the example provided install packages with `yum` which requires superuser permissions on the container.

We can either install the chapter with a more simple example, building a container with Python packages (no need to use `sudo`).  And then continue with the second example using superuser permissions.

### PR DESCRIPTION
**Pull request by michmx**
Example of building a container without sudo using a Python image and Uproot.

### CODE REVIEW

File: _episodes/04-building-containers.md [hunk 1]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
-# Build a container in a interactive session
-
-For building a container from a existing image in an interactive way, `build` provides a flag `--sandbox` that will
-create a writable directory.
-
-~~~bash
-singularity build --sandbox myCentOS7 docker://centos:centos7
+## Build a container in an interactive session
+
+While images contained in the `.sif` files are immutable objects, ideal for reproducibility, for building images in an
```
Comment (amanmdesai) [inline _episodes/04-building-containers.md]: line 30-31  -- looks like something is missing
maybe update them?

File: _episodes/04-building-containers.md [hunk 2]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
-The sandbox name is `myCentOS7`, and it has been initialized from the [official Docker image](https://hub.docker.com/_/centos)
-of CentOS7.
-To initialize an interactive session use `shell`. And to write files within the sandbox directory  `--writable`.
-Additionally, the installation of new components will require superuser access:
+## Building a container with superuser permissions
 
-~~~bash
-sudo singularity shell --writable myCentOS7
+Some operations, like the installation of new components will require superuser access. To be superuser inside the container,
+you must be also outside (in the host machine). You can either open a shell as root, or use `sudo`:
```
Comment (amanmdesai) [inline _episodes/04-building-containers.md]: you must be superuser outside (in the host machine)

File: _episodes/04-building-containers.md (hunk 1 above)
Comment (klieret) [inline _episodes/04-building-containers.md]: Hard parse

```
Images contained in the `.sif` files are immutable objects, ideal for reproducibility. However, it is convenient to build images interactively. 
The command `build` ...
```

File: _episodes/04-building-containers.md [hunk 3]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
+While images contained in the `.sif` files are immutable objects, ideal for reproducibility, for building images in an
+interactive session is convenient the usage of a _sandbox_.
+The command `build` provides a flag `--sandbox` that will create a writable directory:
+```bash
+singularity build --sandbox myPython docker://python:3.9
+```
+The container name is `myPython`, and it has been initialized from the [Docker image](https://hub.docker.com/_/python)
+of Python using the tag 3.9.
+
+To initialize an interactive session use `shell`. And to write files within the container use the flag `--writable`:
```
Comment (klieret) [inline _episodes/04-building-containers.md]: session, use 
(...)
container, use

File: _episodes/04-building-containers.md [hunk 4]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
+```bash
+singularity shell --writable myUbuntu
```
Comment (klieret) [inline _episodes/04-building-containers.md]: wasn't it `myPython`?

File: _episodes/04-building-containers.md (hunk 3 above)
Comment (klieret) [reply _episodes/04-building-containers.md]: Or better "Add the `--writable` flag to modify flags within the container."

File: _episodes/04-building-containers.md [hunk 5]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
 
-The sandbox name is `myCentOS7`, and it has been initialized from the [official Docker image](https://hub.docker.com/_/centos)
-of CentOS7.
-To initialize an interactive session use `shell`. And to write files within the sandbox directory  `--writable`.
-Additionally, the installation of new components will require superuser access:
+## Building a container with superuser permissions
 
-~~~bash
-sudo singularity shell --writable myCentOS7
+Some operations, like the installation of new components will require superuser access. To be superuser inside the container,
```
Comment (klieret) [inline _episodes/04-building-containers.md]: comma: operations, like (...), will

File: _episodes/04-building-containers.md (hunk 2 above)
Comment (klieret) [inline _episodes/04-building-containers.md]: you must also be a superuser outside

File: _episodes/04-building-containers.md [hunk 6]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
+you must be also outside (in the host machine). You can either open a shell as root, or use `sudo`:
 
+```bash
+$ sudo singularity shell --writable myCentOS7
```
Comment (klieret) [inline _episodes/04-building-containers.md]: I think we removed the dollars everywhere else, see #41

File: _episodes/04-building-containers.md [hunk 7]
```diff
@@ -5,61 +5,80 @@ exercises: 30
...
 ~~~
 {: .output}
 
-As an example, let's create a container with Pythia8 available using the `myCentOS7` sandbox.
+As a second example, let's create a container with Pythia8 available using the `myCentOS7` sandbox.
 First, we need to install the development tools (remember that in this interactive session we are superuser):
-
-~~~bash
+```bash
 Singularity> yum groupinstall 'Development Tools'
```
Comment (klieret) [inline _episodes/04-building-containers.md:82]: btw, is `yum` explained somewhere? Not everyone might be familiar with that

**Review by klieret — APPROVED** [review]
Nice! Sorry for the many nitpicky comments

File: _episodes/04-building-containers.md (hunk 7 above)
Comment (michmx) [reply _episodes/04-building-containers.md:82]: Good point! I will add a bit of information.

File: _episodes/04-building-containers.md (hunk 1 above)
Comment (michmx) [reply _episodes/04-building-containers.md]: Ok, changing as suggested.

File: _episodes/04-building-containers.md (hunk 4 above)
Comment (michmx) [reply _episodes/04-building-containers.md]: Oh yes 😅

**Comment by michmx** [comment]
@klieret all comments addressed. Do you think this is ready to go?

**Pull request merged**